src/main/java/com/avengers/gamera/constant/ArticleType.java

```diff
@@ -0,0 +1,26 @@
+package com.avengers.gamera.constant;
+
+import java.util.Objects;
+import java.util.stream.Stream;
+
+public enum ArticleType {
+    NEWS("news"),
+    REVIEWS("reviews");
+
+    private final String typeName;
+
+    ArticleType(String typeName) {
+        this.typeName = typeName;
+    }
+
+    public String getTypeName() {
+        return this.typeName;
+    }
+
+    public static ArticleType of(String name) {
+        return Stream.of(ArticleType.values())
+                .filter(p -> Objects.equals(p.getTypeName(), name))
+                .findFirst()
+                .orElseThrow(IllegalArgumentException::new);
```

Comment:
If you use method reference here, what error message would be along with the exception?

Reply:
Besides, if we throw an exception here, do we have exceptions handler to return the corresponding response? Also, error should be also logged for debugging.

Reply:
Error handler later....  ( TODO